src/backend/distributed/utils/shardinterval_utils.c

```diff
@@ -442,6 +433,43 @@ SearchCachedShardInterval(Datum partitionColumnValue, ShardInterval **shardInter
 }
 
 
+/*
+ * CalculateUniformHashRangeIndex returns the index of the hash range in
+ * which hashedValue falls, assuming shardCount uniform hash ranges.
+ *
+ * We use 64-bit integers to avoid overflow issues during arithmetic.
+ */
```

Comment:
I sometimes find these functions a  little dangerous as someone might  call this function instead of `FindShardIntervalIndex`. So, maybe add this comment:
```
... assuming shardCount uniform hash ranges. In other words, do not rely on this function for finding shard index for all cases.
```

Also, unrelated to this patch, we already seem to have `shardInterval->shardIndex`, why  do we calculate again? There should be a reason :D

Reply:
Also this function is static whereas `FindShardIntervalIndex` is not, so it makes it less like to misuse(other than misusages in the same file).

Reply:
Well, making it `external` is pretty simple  :) We don't have a good distinction in terms of public/static functions overall..

Reply:
Yeah that is also true :)